### PR TITLE
[DO NOT MERGE] chore(Constants): Add guild_subscriptions option

### PR DESCRIPTION
**Pull request by MrJacz**
**Please describe the changes this PR makes and why it should be merged:**
Adds guild_subscriptions from https://github.com/discordapp/discord-api-docs/pull/1016 also changes large_threshold to 50 as that is the default set by discord

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Constants.js [hunk 1]
```diff
@@ -58,13 +58,15 @@ exports.DefaultOptions = {
...
    * @typedef {Object} WebsocketOptions
-   * @property {number} [large_threshold=250] Number of members in a guild to be considered large
+   * @property {number} [large_threshold=50] Number of members in a guild to be considered large
    * @property {boolean} [compress=false] Whether to compress data sent on the connection
+   * @property {boolean} [guild_subscriptions=true] enables dispatching of guild subscription events (presence and typing events)
    * (defaults to `false` for browsers)
    */
   ws: {
-    large_threshold: 250,
+    large_threshold: 50,
```
Comment (vladfrangu) [inline src/util/Constants.js]: Why is this change done?

File: src/util/Constants.js [hunk 2]
```diff
@@ -58,13 +58,15 @@ exports.DefaultOptions = {
...
     compress: false,
+    guild_subscriptions: true,
```
Comment (vladfrangu) [inline src/util/Constants.js]: This defaults to true, but this change would bring FAR too many (possibly breaking) changes to discord.js due to all the events that get stopped.

File: src/util/Constants.js [hunk 3]
```diff
@@ -58,13 +58,15 @@ exports.DefaultOptions = {
   /**
    * WebSocket options (these are left as snake_case to match the API)
    * @typedef {Object} WebsocketOptions
-   * @property {number} [large_threshold=250] Number of members in a guild to be considered large
+   * @property {number} [large_threshold=50] Number of members in a guild to be considered large
    * @property {boolean} [compress=false] Whether to compress data sent on the connection
+   * @property {boolean} [guild_subscriptions=true] enables dispatching of guild subscription events (presence and typing events)
```
Comment (vladfrangu) [inline src/util/Constants.js]: This documentation string is incomplete.

nit 1. Capitalize Enables

As you can see at https://github.com/discordapp/discord-api-docs/pull/1016#issuecomment-510492900, turning this off right now disables... GUILD_MEMBER_*.. That could potentially break expected library behavior, or worst case scenario, the library as a whole due to the cache not being kept up to date. Also, This is probably not how we want to expose such a WS setting. It'd probably become a client setting that is linked to the ws one, with proper documentation, disclaimers, TESTING, etc.

File: src/util/Constants.js (hunk 2 above)
Comment (advaith1) [reply src/util/Constants.js]: true is the default, false disables it. normally all bots have this set to true.

Comment (MrJacz) [reply src/util/Constants.js]: yes. what are you suggesting i change exactly?

Comment (advaith1) [reply src/util/Constants.js]: I think vlad thought true meant disable subscriptions

Comment (vladfrangu) [reply src/util/Constants.js]: That was a bad wording on my part, and I'm sorry. I know it defaults to true, and I meant that turning the flag off would bring far too many possibly breaking changes! 😅

Comment (advaith1) [reply src/util/Constants.js]: Isn't the ability to turn it off the whole point of this PR?

**Comment by Monbrey** [comment]
As per the link Vlad provided up there, even though turning it off would be optional the impact of doing so is much larger than just disabling PRESENCE_UPDATE and TYPING_START, which is what the Discord API docs suggest it will do.

It effectively turns off almost all updates for GuildMembers, including when they're added or removed from the Guild, and when their roles change.

**Comment by appellation** [comment]
I think this is very appropriate to keep in the ws identify options. Using this feature is highly advanced and there is no way for us to fully document its effects without some extremely thorough testing that I doubt will get done. I would suggest that using this feature is event more advanced than disabling events, where we explicitly state that the behavior is undefined. That being said, I'm definitely on-board with clearly documenting that disabling guild subscriptions will break significant portions of the library.

**Comment by NightYoshi370** [comment]
> Thank you! But, for now, this PR should be kept as pending till we see what changes (if anything) with the guild_subscriptions boolean. This would also be exposed as a separate client option that gets tied to the ws one, and, as said in my review, would include faar bigger warnings and such! +1

The Pull Request mentioned in OP is merged. Heck, if its that new webhook news to channel thing, I think it's in the client itself.

**Comment by vladfrangu** [comment]
> The Pull Request mentioned in OP is merged. 

That it is, doesn't change the fact this is an option that can come with a LOT of side effects in discord.js, most of which we can't test

> Heck, if its that new webhook news to channel thing, I think it's in the client itself

It isn't

**Comment by Rexogamer** [comment]
question: they updated the maximum amount of boosts (which I think this is about) to 30; does anything in this pr need updating?

File: src/util/Constants.js (hunk 3 above)
Comment (NotSugden) [inline src/util/Constants.js]:
```suggestion
   * @property {boolean} [guild_subscriptions=true] enables dispatching of guild subscription events
   * (presence and typing events)
```
line length was over 120, breaking the max-line-length rule

**Comment by iCrawl** [comment]
https://discordapp.com/developers/docs/topics/gateway#guild-subscriptions

**Pull request closed without merging**